### DISCUSSION TITLE
--recursive flag doesn't seem to run in child directories

### DISCUSSION THREAD

**Discussion by caoimheharvey** [terraform-linters/tflint#2133] [Q&A]
### Summary

I have my `.tflint.hcl` config in a parent directory and run `tflint --init` in the parent directory, and I have a lot of child directories that I want to lint, but when I run `tflint --recursive --config="$(pwd)/.tflint.hcl"` from the parent directory it returns nothing even if there are linting failures. 
When I switch to debug output this is what I am seeing:
```
│ Failed to run in child/nested/nested/nested; exit status 1
Failed to initialize plugins; Plugin "terraform" not found. Did you run "tflint --init"?
│ Failed to run in child; exit status 1
Failed to initialize plugins; Plugin "terraform" not found. Did you run "tflint --init"?
│ Failed to run in child/nested; exit status 1
Failed to initialize plugins; Plugin "terraform" not found. Did you run "tflint --init"?
│ Failed to run in child/nested; exit status 1
```
Seems odd that I would need to run `tflint --init --recursive` because I would very quickly hit a rate limit (tried this), so its not a practical solution. 
As a work-around I've tried to copy the `.tflint.d` directory to all my subdirectories but again, this isn't a workable solution. 

### Command

tflint --recursive --config="$(pwd)/.tflint.hcl"

### Terraform Configuration

```terraform
Not a terraform specific issue as far as I can tell
```

### TFLint Configuration

```HCL
config {
  plugin_dir = "./.tflint.d/plugins"
}

plugin "terraform" {
  enabled = true
  version = "0.9.0"
  source  = "github.com/terraform-linters/tflint-ruleset-terraform"
}

plugin "aws" {
  enabled = true
  version = "0.33.0"
  source  = "github.com/terraform-linters/tflint-ruleset-aws"
}
```

### Output

```Shell
│ Failed to run in child/nested/nested/nested; exit status 1
Failed to initialize plugins; Plugin "terraform" not found. Did you run "tflint --init"?
│ Failed to run in child; exit status 1
Failed to initialize plugins; Plugin "terraform" not found. Did you run "tflint --init"?
│ Failed to run in child/nested; exit status 1
Failed to initialize plugins; Plugin "terraform" not found. Did you run "tflint --init"?
│ Failed to run in child/nested; exit status 1
```
```


### TFLint Version

0.53.0

### Terraform Version

1.9.5

### Operating System

- [X] Linux
- [X] macOS
- [ ] Windows

**Comment by bendrucker** [comment]
This is the documented behavior:

https://github.com/terraform-linters/tflint/blob/be128251367e75723c01e666a9f675400317a4d4/docs/user-guide/plugins.md#plugin-directory

If you pass a relative directory, that's relative to the working directory, not to the config file. Perhaps it should be otherwise, but that would be breaking too for users relying on it working as documented currently.

If you just let the plugins get managed in your home directory like they are by default, you'd be fine. Otherwise, you have to set `TFLINT_PLUGIN_DIR` to an absolute path just like you do for the configuration to prevent paths from being resolved relative to the working directory, i.e. your child modules.

**Reply by caoimheharvey** [reply]
I've tried updating to use the absolute path instead of the relative path and I'm getting: 
```
$ cd folder/folder
$ tflint --init
Installing "terraform" plugin...
Installed "terraform" (source: github.com/terraform-linters/tflint-ruleset-terraform, version: 0.9.0)
Installing "aws" plugin...
Installed "aws" (source: github.com/terraform-linters/tflint-ruleset-aws, version: 0.33.0)
$ ls -lha
total 36K    
drwxrwxrwx    7 root     root        4.0K Oct  7 10:05 .
drwxrwxrwx   20 root     root        4.0K Oct  7 10:05 ..
drwxr-xr-x    3 root     root        4.0K Oct  7 10:05 .tflint.d
-rw-rw-rw-    1 root     root         486 Oct  7 10:05 .tflint.hcl
$ tflint --recursive
│ Failed to run in folder; exit status 1
Failed to initialize plugins; Plugin "terraform" not found. Did you run "tflint --init"?
│ Failed to run in folder; exit status 1
Failed to initialize plugins; Plugin "terraform" not found. Did you run "tflint --init"?
```
This is the config in the hcl file:
```
config {
  plugin_dir = "/builds/namespace/repo/folder/folder/.tflint.d/plugins"
}
```
